src/network/networkctl.c

```diff
@@ -1379,6 +1381,32 @@ static int show_logs(const LinkInfo *info) {
                         NULL);
 }
 
+static int show_cloud_provider(const LinkInfo *info, Table *table) {
+        _cleanup_strv_free_ char **private_ips = NULL, **public_ips = NULL;
+        int r;
+
+        assert(info);
+
+        if (detect_virtualization() == VIRTUALIZATION_MICROSOFT) {
```

Comment:
Would it be feasible to avoid provider-specific code paths in networkctl, and instead have that logic in the providerd implementation only?

Reply:
I see from various providers they have different attributes. Hence it provider specific APIS. I would guess user should we know which platform they are using.